### PR TITLE
fix: load normal envs during build

### PR DESCRIPTION
**Pull request by keeganpotgieter**
### TL;DR

Removed the creation of the `.env.production` file from the GitHub Actions release workflow, and updated the Rollup configuration to use environment-specific `.env` files based on the `NODE_ENV` variable.

### What changed?

- Removed the creation of `.env.production` file in `.github/workflows/release.yml`.
- Updated `rollup.config.mjs` to load environment-specific `.env` files based on `NODE_ENV`.

### How to test?

1. Trigger the GitHub Actions release workflow and ensure it skips the creation of `.env.production` file.
2. Verify that the build process works correctly and environment variables are loaded based on the `NODE_ENV` setting.

### CODE REVIEW

**Comment by keeganpotgieter** [comment]
* **#84** <a href="https://app.graphite.dev/github/pr/subifinancial/subi-connect/84?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @keeganpotgieter and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
